### ISSUE TITLE
Bug with trim()

### ISSUE THREAD

**Issue by itlogicllc** [twigphp/Twig#2400]
I am using the trim() function with _self to get the current page without the .twig extension. The code looks like this:

`{% set active = _self|trim('.twig') %}`

It works fine except when the filename of the page ends with a 'T'. It that case the 'T' in the file name before the '.' gets truncated. It seems that the 'T' in the file name is getting truncated because the extension also starts with 'T'.

In other words, if the file name of the template were mypage.twig or home.twig, the above snippet would be:

active = mypage
active = home

If the file name was test.twig or contact.twig, the snippet would be:

active = tes
active = contac

Perhaps this is a bug?

**Comment by jfcherng** [comment]
This is the same behavior with the [trim](http://php.net/manual/en/function.trim.php) function in PHP.
